CRM/Civicase/Form/CaseSalesOrderContributionCreate.php

```diff
@@ -163,15 +162,10 @@ public function validateAmount(array $values) {
       ->jsonSerialize();
 
     $paidTotal = array_sum(array_column($caseSalesOrderContributions, 'contribution_id.total_amount'));
-    $paidPercent = ($paidTotal * 100) / $caseSalesOrder['total_after_tax'];
-    $remainPercent = 100 - $paidPercent;
-
-    if ($remainPercent <= 0) {
-      $errors['to_be_invoiced'] = 'Contribution amount cannot exceed the total sales order amount';
-    }
+    $remainBalance = $caseSalesOrder['total_after_tax'] - $paidTotal;
 
-    if ($values['to_be_invoiced'] == self::INVOICE_PERCENT && floatval($values['percent_value']) > $remainPercent) {
-      $errors['percent_value'] = 'Percentage value cannot exceed ' . round($remainPercent, 2);
+    if ($remainBalance <= 0) {
+      $errors['to_be_invoiced'] = 'No remaining balance to create contribution for.';
```

Comment:
maybe "Unable to create a contribution due to insufficient balance." is more clear message